activerecord/lib/active_record/railtie.rb

```diff
@@ -148,10 +148,7 @@ class Railtie < Rails::Railtie # :nodoc:
           ActiveSupport.on_load(:active_record) do
             db_config = ActiveRecord::Base.configurations.configs_for(env_name: Rails.env).first
```

Comment:
@rafaelfranca we just ran into a version of this issue, and came across this PR, and I'm curious whether this part of the Railtie also needs to be changed.

Right now it only is loading the schema cache for the first DB, but the rest of your PR seems to indicate that other DBs *are* now supported.

Reply:
It doesn't need to be changed, it just need to be removed. As it is it works, but it is just duplicated work. It now works for all the databases, even if this code only load the first db.

I removed this code in https://github.com/rails/rails/pull/51034

Reply:
Ohh even better! Thanks for the quick response @rafaelfranca.